### DISCUSSION TITLE
Evaluating Mapsui WPF capability

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [Mapsui/Mapsui#2752] [Q&A]
Have a start on a `SKElement` based cartography problem. I have an `ICartographer` in my DI container doing the basics, preparing a baseline `SKBitmap` for use with the `SKElement` getting started.

![image](https://github.com/user-attachments/assets/3c08422e-1383-4384-b2b9-d230b95d9cc1)

I wonder, can I provide the `Mapsui` control(s) this instance, `SKBitmap` or `SKImage` would be fine?

In addition to which, the value add would be fantastic if it also provided:

1. Automatically negotiating scaling, including zoom pan, size changed scenarios.
1. Keyboard mouse event handling, grab panning, mouse wheel zoom.
1. Automatic bounds checking i.e. the image should not be grabbed or zoomed beyond some sensible scaling restrictions.
1. Ability to click on the map image and identify some coordinates, which we can take and later transform for use in terms of world coordinates, i.e. from canvas to screen to world.
1. Customization of Mapsui layers, somethings probably are not necessary, perhaps things like an on screen scale, something like that, but much beyond that, do not need fancy layers.

To name a few obvious MVP 'min-viable-prod' features I can think of. Wonder if `Mapsui` does this already, would certainly save me some time.

**Comment by mwpowellhtx** [comment]
Just out of the box, cloned running the WPF sample, the basic level of control is there, pan, zoom. Do not need to source any street maps, anything like that; also do not need to rotate, far from it. Might like to somehow engage with the map, click for locations, transformed how we need to, i.e. canvas to screen and or then to world, of course, in some way shape form how we receive the point, coordinate, etc. Do not necessarily need layering assists, things like tiling etc; like I explained, we are doing the basic prep work on the image itself, and just need the functional viewer.

**Comment by inforithmics** [comment]
Here is a "MapControl" for only creating Images for Regression Testing of Mapsui.

https://github.com/Mapsui/Mapsui/blob/main/Tests/Mapsui.Rendering.Skia.Tests/RegressionMapControl.cs

**Reply by mwpowellhtx** [reply]
Not hundred percent certain, what is it I am looking at there? Although I gather in some way `_map`, perhaps `_map.Navigation`, is somehow the nexus point. How do I provide my `SKBitmap`? Which to be clear, I do not need for `Mapsui` to create anything; I have the already created `SKBitmap` already `SkiaSharp` prepared; just need the viewer aspect.

**Reply by inforithmics** [reply]
All Sample code can be viewed here, To see which capabilities mapsui has.
https://mapsui.com/samples/
